### ISSUE TITLE
[java] UncommentedEmptyConstructor reports protected/package-private constructors

### ISSUE THREAD

**Issue by mkordas** [pmd/pmd#934]
Please, prefix the report title with the language it applies to within brackets, such as [java] or [apex].
If not specific to a language, you can use [core].

**Affects PMD Version:** 
6.0.1

**Rule:**
UncommentedEmptyConstructor

**Description:**
UncommentedEmptyConstructor reports protected/package-private constructors in public classes, while they are usually intentional to limit accesibility to constructor. IMO only empty constructors that have the same visibility as class should be reported.

**Code Sample demonstrating the issue:**
The following should not be the issue:
```java
public class Event {
	protected Event() {
	}
}
```
The following should remain the issue:
```java
class Event {
	Event() {
	}
}
```

**Comment by adangel** [comment]
@mkordas 
You are right: the rule currently reports any empty constructor, that is not private.
However, I would consider enforcing the comments as a matter of taste. It's simple and sometimes useful to add an explanatory comment, when needed.

I'd suggest to enhance the rule and add a property with which the behavior of the rule can be controlled, such as a "strict" property. What do you think?

**Comment by mkordas** [comment]
@adangel yes, good idea. Property would be really useful, as the current behavior is too strict for me (there are self-documenting cases).
